### PR TITLE
Closes #2186 TERMID -> TERMNUM, TERMNAME -> TERMCHAR

### LINKED ISSUE #2186: Feature Request:  `TERMID`/`TERMNAME` should be renamed to `TERMNUM` and `TERMCHAR`
### Feature Idea

https://github.com/pharmaverse/admiral/discussions/2010#discussioncomment-6916013

With respect to query-related functions in admiral, swapping out

`TERMID` for `TERMNUM` 
`TERMNAME` for `TERMCHAR` 

may be more informative/lead to less confusion

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by manciniedoardo**
Closes #2186 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
[This PR](https://github.com/pharmaverse/pharmaversesdtm/pull/77) will need to be merged in pharmaversesdtm to fix the CICD error.

**Review by bms63 — COMMENTED** [review]
Do any of the templates need updating?

**Comment by manciniedoardo** [comment]
> Do any of the templates need updating?

I searched through the package and couldn't find anything.

File: vignettes/occds.Rmd [hunk 1]
```diff
@@ -456,12 +456,12 @@ Standardized Drug Groupings (SDG).
...
-  "SDG01",     "Diuretics",        11,  "BROAD",       1,  "CMDECOD",   "Diuretic 2",           NA,
-  "SDG02",     "Costicosteroids",  12,  "BROAD",       1,  "CMDECOD",   "Costicosteroid 1",     NA,
-  "SDG02",     "Costicosteroids",  12,  "BROAD",       1,  "CMDECOD",   "Costicosteroid 2",     NA,
-  "SDG02",     "Costicosteroids",  12,  "BROAD",       1,  "CMDECOD",   "Costicosteroid 3",     NA,
+  ~PREFIX, ~GRPNAME, ~GRPID, ~SCOPE, ~SCOPEN, ~SRCVAR, ~TERMCHAR, ~TERMNUM,
+  "SDG01", "Diuretics", 11, "BROAD", 1, "CMDECOD", "Diuretic 1", NA,
+  "SDG01", "Diuretics", 11, "BROAD", 1, "CMDECOD", "Diuretic 2", NA,
+  "SDG02", "Costicosteroids", 12, "BROAD", 1, "CMDECOD", "Costicosteroid 1", NA,
+  "SDG02", "Costicosteroids", 12, "BROAD", 1, "CMDECOD", "Costicosteroid 2", NA,
+  "SDG02", "Costicosteroids", 12, "BROAD", 1, "CMDECOD", "Costicosteroid 3", NA,
```
Comment (bundfussr) [inline vignettes/occds.Rmd:464]: Could you preserve the alignment?

Comment (manciniedoardo) [reply vignettes/occds.Rmd:464]: It's impossible to align the other columns (e.g. GRPID is unaligned anyway) so how do you suggest doing this?

Comment (bundfussr) [reply vignettes/occds.Rmd]: You are right, it was not completely aligned before. Strange that it passed the styler check before.

I have aligned them. Please have a look.

Comment (manciniedoardo) [reply vignettes/occds.Rmd]: Ah I see thanks - looks good :smile:

**Comment by bundfussr** [comment]
@bms63 , could you have a look and merge it if it is fine with you?

Merging this PR should fix all the failing workflows in admiral.

**Pull request merged**